b2sdk/encryption/provider.py

```diff
@@ -0,0 +1,65 @@
+######################################################################
+#
+# File: b2sdk/encryption/provider.py
+#
+# Copyright 2021 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+from abc import ABCMeta, abstractmethod
+from typing import Dict, Optional
+
+from ..file_version import FileVersionInfo
+from .setting import EncryptionSetting
+
+
+class AbstractEncryptionSettingsProvider(metaclass=ABCMeta):
```

Comment:
I would add abstract init

Reply:
it's not actually needed, you can have a class with hardcoded keys that won't have an init

Reply:
It would be beneficial for future compatibility. An init with *args, **kwargs. When you add a new parameter to init in the future, it won't be NBC. I would do the same with every API - add *args and **kwargs